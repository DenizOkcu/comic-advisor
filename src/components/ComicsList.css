.comics-list {
  flex: 1;
  height: 100%;
  margin-right: 10px;
  overflow-y: auto;
}

.bottom-nav {
  display: flex;
  flex-direction: row;
}

.random-button,
.recent-button {
  flex: 1;
  background-color: red;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  display: inline-block;
  font-family: "Bangers", Arial Black;
  font-size: 2em;
  margin: 0 5px 10px 0;
  padding: 0.05em 0;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  width: 100%;
}

.random-button:last-child,
.recent-button:last-child {
  margin-right: 0;
}
