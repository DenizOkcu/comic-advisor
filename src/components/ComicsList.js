import React from "react";
import ComicsListItem from "./ComicsListItem";

// im rest destrukturieren (kann auf default) { comics, selectComic = {} }
const ComicsList = ({ comics, selectComic }) => {
  // kann auch in zeile 18 passieren
  const listItems =
    comics &&
    comics.map(comic => (
      <ComicsListItem key={comic.id} comic={comic} selectComic={selectComic} />
    ));

  return listItems ? (
    <aside className="four columns">
      <h4>Comics List</h4>
      <hr />
      {listItems}
    </aside>
  ) : (
    ""
  );
};

export default ComicsList;
