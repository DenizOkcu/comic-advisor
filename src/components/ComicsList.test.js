import React from "react";
import { shallow } from "enzyme";
import ComicsList from "./ComicsList";

describe("ComicsList", () => {
  it('should render correctly in "debug" mode', () => {
    const comics = [
      {
        id: 1,
        creators: [{ name: "writer name" }],
        thumbnail: { path: "test", extension: "jpg" },
        prices: [{ price: 3.99 }],
        title: "title",
        pageCount: 32
      }
    ];
    const component = shallow(<ComicsList comics={comics} debug />);

    expect(component).toMatchSnapshot();
  });
});
