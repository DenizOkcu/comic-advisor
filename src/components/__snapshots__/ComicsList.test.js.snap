// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicsList should render correctly in "debug" mode 1`] = `
<aside
  className="four columns"
>
  <h4>
    Comics List
  </h4>
  <hr />
  <ComicsListItem
    comic={
      Object {
        "creators": Array [
          Object {
            "name": "writer name",
          },
        ],
        "id": 1,
        "pageCount": 32,
        "prices": Array [
          Object {
            "price": 3.99,
          },
        ],
        "thumbnail": Object {
          "extension": "jpg",
          "path": "test",
        },
        "title": "title",
      }
    }
    key="1"
  />
</aside>
`;
